test/testVectorTypesTrackPropagationPipeLine.cxx

```diff
@@ -76,15 +76,18 @@ template <typename Data> struct Tracks_v {
 
   // Gather info to fill one SIMD lane
   void Gather(Data *track, const std::size_t lane) {
-    Set(fPos_v.x(), lane, track->Position().x());
-    Set(fPos_v.y(), lane, track->Position().y());
-    Set(fPos_v.z(), lane, track->Position().z());
-    Set(fDir_v.x(), lane, track->Direction().x());
-    Set(fDir_v.y(), lane, track->Direction().y());
-    Set(fDir_v.z(), lane, track->Direction().z());
-    Set(fCharge_v, lane, track->Charge());
+    vecgeom::Vector3D<double> pos = track->Position();
+    vecgeom::Vector3D<double> dir = track->Direction();
+    
+    Set(fPos_v.x(),  lane, pos.x());
+    Set(fPos_v.y(),  lane, pos.y());
+    Set(fPos_v.z(),  lane, pos.z());
+    Set(fDir_v.x(),  lane, dir.x());
+    Set(fDir_v.y(),  lane, dir.y());
+    Set(fDir_v.z(),  lane, dir.z());
+    Set(fCharge_v,   lane, track->Charge());
     Set(fMomentum_v, lane, track->P());
-    Set(fNSteps_v, lane, track->GetNsteps());
+    Set(fNSteps_v,   lane, track->GetNsteps());
```

Comment:
Is there a noticeable difference in the profile?

Reply:
Hi Guilherme, I cannot see yet a difference in my profile since, in my machine, most of the time is spent in the SinCos Vc implementation.